File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        l,r= 0,1
        count = 0
        while(l<r):
            if r<n and nums[l]+nums[r] == target:
                return [l,r]
            else:
                count+=1
                r+=1
            if count == n-1:
                l+=1
                r = l+1
                count = 0

File: Data Structures & Algorithms/concatenation-of-array/submission-0.rs
impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
        let mut ans = Vec::new();
        let n = nums.len();
        
        for i in 0..2*n{
            ans.push(nums[i%n]);
        }

        ans
    }
}
